index.js

```diff
@@ -1,120 +1,30 @@
 app.use("/public", express.static(`${process.cwd()}/public`));
 
-app.get("/", function (req, res) {
-  res.sendFile(process.cwd() + "/views/index.html");
-});
-
-const URLSchema = mongoose.Schema(
-  {
-    longURL: String,
-    shortURL: String,
-    issuedDate: Date,
-  },
-  { versionKey: false }
-);
-
-const URL = mongoose.model("URL", URLSchema);
-
-//Might need to update the RNG with another library
-function shorten() {
-  const characters =
-    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
-  let shortURL = "";
-  for (i = 0; i < 8; i++) {
-    shortURL += characters[Math.floor(Math.random() * characters.length)];
-  }
-  return shortURL;
-}
-
-async function removeOldEntries() {
-  const thirtyDaysAgo = Date.now() - 2592000000;
-  try {
-    const deleteObj = await URL.deleteMany({
-      issuedDate: { $lte: thirtyDaysAgo },
-    });
-    console.log(`Number of documents deleted: ${deleteObj.deletedCount}`);
-  } catch (err) {
-    console.error("Error: ", err);
-  }
-}
-
-//Post URL (check if its already in the collection; if YES return and extend timer), convert and save to MongoDB, return JSON
-
-app.post("/api/shorturl", async (req, res) => {
-  removeOldEntries();
-  const url = req.body.url;
-
-  //Would replace this with client-side form validation in the future
-  if (!url.includes("https://") && !url.includes("http://")) {
-    return res.json({ error: "invalid url" });
-  }
-
-  try {
-    const checkObject = await URL.findOne({ longURL: url });
-    if (checkObject?.shortURL) {
-      checkObject.issuedDate = Date.now();
-      await checkObject.save();
-      res.json({ shortURL: checkObject.shortURL });
-    } else {
-      const newObject = new URL({
-        longURL: url,
-        shortURL: shorten(),
-        issuedDate: Date.now(),
-      });
-      await newObject.save();
-      res.json({ shortURL: newObject.shortURL });
-    }
-  } catch (err) {
-    res.send("Error: " + err);
-  }
+// Get DB health
+app.get("/db-health", (req, res) => {
+  res.json({ status: connState() });
 });
 
-// add /api/get-link endpoint to read 'http://service-domain/{publicId}', find mongo entry by checking the id, and then respond the real link
-
-app.get("/api/get-link/:shorturl", async (req, res) => {
-  const shorturl = req.params.shorturl;
-  try {
-    const checkObject = await URL.findOne({ shortURL: shorturl });
-    res.json({ "Long URL": checkObject.longURL });
-  } catch (err) {
-    res.send(err);
-  }
+// Serve index page
+app.get("/", function (req, res) {
+  res.sendFile(process.cwd() + "/views/index.html");
 });
 
-// add another endpoint that instead od responding string, will execute 302 redirect. Map it to 'service-domain/l/publicId'
-
-app.get("/l/:publicId", async (req, res) => {
-  const shorturl = req.params.publicId;
-  try {
-    const checkObject = await URL.findOne({ shortURL: shorturl });
-    res.redirect(checkObject.longURL);
-  } catch (err) {
-    res.send(err);
-  }
-});
+apiRoutes(app)
```

Comment:
What is going on here and how does it work?

Reply:
I see that you've refactored the routes into a single function but I can't really envision how it operates in terms of the function (apiRoutes) calling the Express server in index.js (apiRoutes(app)).

Reply:
It's the same as before, except that I moved that 'iterative' part of defining the routes into a function. 
In JS you can pass any type of object to a function, so I pass reference of the express app instance ('app'), and then using that reference to define the routes in the routes function.

It the api-routes.js there is the export part `module.exports = routes` which means that I export the routes function as default from that module. Hence calling this one `const apiRoutes = require('./src/api-routes')` will actually give me the reference to the default fn exported, or this `const routes = function(app) {`